src/store/lokapi.ts

```diff
@@ -88,6 +89,18 @@ export var moduleLokAPI = {
     async fetchUserAccountValidationRights({ commit, state }:any) {
       let hasRight = await lokApiService.hasUserAccountValidationRights()
       commit('setHasUserAccountValidationRights', hasRight)
+    },
+    async fetchPendingUserAccounts({ commit, state }:any) {
+      let accounts = await lokApiService.getStagingUserAccounts()
+      commit('setPendingUserAccounts', accounts)
+    },
+    async validateUserAccount({ commit, state, dispatch }:any, userAccount: any) {
+      let account = state.pendingUserAccounts.find((account: any) => account.id === userAccount.id)
+      if (typeof account !== 'undefined') {
+        let result = await account.validateCreation()
+        dispatch('fetchPendingUserAccounts')
+        return result
+      }
```

Comment:
The thing is, you tied up early the workings of what we call now lokApiService with the store. So this keeps the code in line with this choice.
Moreover, I personally like not to scatter responsibilities everywhere and keep things tidy (as you like to have you pretty tidy commit messages so your parser will read them). So I think the extra lines of code are worth it in order to keep all things interacting with the lokapi in the same place. It is really not a problem to treat the lokapi store as a layer above the lokapi and in fact, that's what it already is. For me the error here is passing complete objects down into the components. They should have been pure-data only.

Reply:
Well, conceptually I do not agree with you on this point (using full blown object, instead of using pure-data here) that this is bad, in my current point of view. There's probably an interesting discussion to do on this topic, if you can give arguments and listen to mine. You'll maybe be surprised to discover that I'm far from being 100% sure on my take on this, even if (to the moment) it is still the option that has my favor. But is this the time to have this talk ? Probably not to my opinion. Can we agree that : 1 - yes, this is an interesting topic and I'm happy to have a respectful talk about it, and I'm happy to listen to your arguments, 2 - no, this is not the right time to talk about that... Then what to think about adding unnecessary code and complexity here to make your point not using the benefit of the actual solution that you don't like ? The store has many internal mecanism imposing a precise structure and very careful design to be a store (mutation should not be async, there are actions and commits... etc ...), a store here means data that is used by the app, that are directly linked to the state of the app. It is here to avoid having to deal with endless callbacks for each component. Using it as a frontend to a service is to add artificially a responsability to the store. Which we may want to consider, or may not. These specific calls here are stateless calls, they do not need to be in the store. I don't see (for now) the added benefit you are talking about, and that this is any cleaner or practical. I still see the added logic and lines that you must add, and the contrived way of doing something that would need 1 line otherwise, without even having to import anything in the view or direct calls to the lokapi. I do not caution this for the previous reasons (that I hope practical, technical, and constructive). But I can live with it.

Reply:
> 2 - no, this is not the right time to talk about that...
> Then what to think about adding unnecessary code and complexity here to make your point not using the benefit of the actual solution that you don't like ?

Sorry @vaab but these two sentences do exactly contradict each other.
If you don't want me to express my opinion just don't ask me to do it in the first place, duh !

You have the code, you have the powers to edit it. Just cut the needless talk and take the decision yourself instead of loosing your and my time.

Reply:
What I will do is halfway between what you wanted and what I'm proposing, that is: make the call to the `validate()` method inside the store, so we can chain the result with the pending accounts list refreshing right from the store, and at the same time, cut the search on the pending list which is, I agree, useless lines of code. That way, in the component, we just have the call to an action, which will validate the account and refresh the list. Once we have the result at the component level, we will fire the notification.

Reply:
Ok I will do how you wanted it to be. After all I might not have your big picture of the design of the lib. I don't even know what the lib does behind the scene: an ajax call ? multiple ajax calls ? anyway it's a promise so it must return a resolve/reject and I will use that to notify error/success and refresh the list if success.